src/lib/github/apiActions/baseApiActions.ts

```diff
@@ -0,0 +1,121 @@
+import { getAccessToken } from "@/lib/auth/getAccessToken";
+
+export type GitHubPagination = {
+  nextPage?: number;
+  firstPage?: number;
+  lastPage?: number;
+  previousPage?: number;
+};
+export type GitHubPaginatedResponse<T> = {
+  data: T;
+  pagination?: GitHubPagination;
+};
+export function BaseGithubApiActions() {
+  const baseUrl: string = "https://api.github.com";
```

Comment:
Specifying the string type is a little redundant because the type can be inferred. This isn't wrong though, you're welcome to leave it if you prefer setting the type explicitly.